web/modules/custom/joinup_group/src/Entity/GroupTrait.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_group\Entity;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\og\OgMembershipInterface;
+
+/**
+ * Reusable methods for group bundles.
+ */
+trait GroupTrait {
+
+  /**
+   * Static cache of memberships for this group, keyed by user ID.
+   *
+   * @var \Drupal\og\OgMembershipInterface[]|NULL[]
+   */
+  protected $memberships;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getMembership(int $uid): ?OgMembershipInterface {
+    assert(is_subclass_of($this, ContentEntityBase::class), __TRAIT__ . ' is intended to be used in bundle classes for content entities.');
+    if (!isset($this->memberships[$uid])) {
+      $og_membership_storage = $this->entityTypeManager()->getStorage('og_membership');
+
+      $query = $og_membership_storage
+        ->getQuery()
+        ->condition('uid', $uid)
+        ->condition('entity_type', $this->getEntityTypeId())
+        ->condition('entity_id', $this->id());
+
+      $membership_ids = $query->execute();
```

Comment:
I did this because of performance reasons, the OG method is inefficient since it loads all the user's memberships and filters them down. But I agree it is not our responsibility to do micro optimization for this. Let's go with the simple solution.

Reply:
I created a ticket in OG to solve the performance issue: https://github.com/Gizra/og/issues/696, I will simplify our code with a call to `MembershipManager::getMembership()`.

Reply:
I have updated the code. I have also removed the static cache since it is also not our responsibility to maintain a per-group membership cache.